lib/config/config.js

```diff
@@ -439,6 +455,101 @@ function loadConfigFile (configFile) {
   return null
 }
 
+/**
+ * Collects information about the agent, environment and configuration options that are
+ * sets from a source (env vars, start options or config file). For each config option it
+ * collects
+ * - source: where te value came from (environment, start, file)
+ * - sourceValue: the raw value provided im the source
+ * - normalizedValue: the value normalized by the configuration which is used in the agent
+ * - file: if source === 'file' this proprty is defined with the path of the config file
+ *
+ * @param {Object} config The configuration object with normalized options
+ * @param {Array<{ name: string, ops: Object}>} sources The list of sources of options ordered by priority
+ * @param {String | undefined} configFilePath Path of the config file if loaded
+ * @returns {Object} Object with agent, environment, and configuration data.
+ */
+function getLoggingPreambleData (config, sources, configFilePath) {
+  const optsFromSources = sources.reduce((prev, s) => Object.assign({}, s.opts, prev), {})
```

Comment:
If I'm not mistaken the in the expression `Object.assign(a, b)`, the var `b` overrides `a` and in this case implicitly will give more priority to the last source from the list. And we want just the opposite.

But your observation made me think about a better API for that. I think we can use `Array.prototype.reduceRight`

Reply:
> I want 1st element to preserve its values since is the one with most priority.

That's why I prefixed with "If you are willing to reverse the priority of the given `sources`".